### ISSUE TITLE
PERF401 false positive when a target variable is global or nonlocal

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#19531]
### Summary

The fix for [`manual-list-comprehension` (PERF401)](https://docs.astral.sh/ruff/rules/manual-list-comprehension/) is incorrect when the `for` loop’s target list includes a global or nonlocal variable. Rewriting the loop with a comprehension means the variable isn’t updated. This is similar to #16445. [Example](https://play.ruff.rs/33af9aba-7285-4899-9777-bc6b14f6c911):
```console
$ cat >perf401.py <<'# EOF'
INDEX = None
def f():
    global INDEX
    result = []
    for INDEX in range(3):
        result.append(+INDEX)
f()
print(INDEX)
# EOF

$ python perf401.py
2

$ ruff --isolated check perf401.py --select PERF401 --preview --unsafe-fixes --fix
Found 1 error (1 fixed, 0 remaining).

$ cat perf401.py
INDEX = None
def f():
    global INDEX
    result = [+INDEX for INDEX in range(3)]
f()
print(INDEX)

$ python perf401.py
None
```

### Version

ruff 0.12.5 (d13228ab8 2025-07-24)

**Comment by IDrokin117** [comment]
@ntBre I am going to work on this
